proxy/sources/rawkuma.py

```diff
@@ -0,0 +1,197 @@
+import json
+import re
+import ast
+
+from bs4 import BeautifulSoup
+from bs4.element import NavigableString
+from django.shortcuts import redirect
+from django.urls import re_path
+
+from ..source import ProxySource
+from ..source.data import ChapterAPI, ProxyException, SeriesAPI, SeriesPage
+from ..source.helpers import api_cache, get_wrapper
+
+class Rawkuma(ProxySource):
+    def get_reader_prefix(self):
+        return "rawkuma"
+
+    def shortcut_instantiator(self):
+        def handler(request, raw_url):
+            slug_name = self.get_slug_name(self.normalize_slug(raw_url))
+            canonical_chapter = self.parse_chapter(raw_url).replace(".", "-")
+            if canonical_chapter:
+                return redirect(
+                    f"reader-{self.get_reader_prefix()}-chapter-page",
+                    slug_name,
+                    canonical_chapter,
+                    "1",
+                )
+            else:
+                return redirect(
+                    f"reader-{self.get_reader_prefix()}-series-page",
+                    slug_name,
+                )
+
+        return [ re_path(r"^rk/(?P<raw_url>[\w\d\/:.-]+)", handler) ]
+
+    @staticmethod
+    def normalize_slug(denormal):
+        normal = re.split(r'-chapter-\d*', denormal.strip('/'))[0]
+        normal = 'https://' + re.sub(r'https?:\/\/', '', normal)
+        return normal
+
+    @staticmethod
+    def get_slug_name(url):
+        return url.strip("/").split("/")[-1]
+
+    @staticmethod
+    def get_series_url(meta_id):
+        return "https://rawkuma.com/manga/" + meta_id
+
+    @staticmethod
+    def parse_chapter(raw_url):
+        search = re.search(r'chapter-([\d-]+)$', raw_url.strip("/"))
+        if not search: return ""
+        return search.group(1).replace("-", ".")
+
+    def rk_scrape_common(self, meta_id):
+        series_url = self.get_series_url(meta_id)
+        resp = get_wrapper(series_url)
+        if resp.status_code == 200:
```

Comment:
(I realize that this is cargo-culted from the other proxies so feel free to ignore) but you could simplify the indentation of these guards by checking the inverse and returning if it's true. Eg.

```python
if resp.status_code != 200:
  return
```

It's just a bit cleaner this way.